stable/airflow/templates/_helpers.tpl

```diff
@@ -84,14 +84,9 @@ Create the name for the airflow secret.
 Map environment vars to secrets
 */}}
 {{- define "airflow.mapenvsecrets" -}}
-    {{- $secretName := .Release.Name | trunc 63 | trimSuffix "-" }}
-    {{- $mapping := .Values.airflow.defaultSecretsMapping }}
-    {{- if .Values.existingAirflowSecret }}
-      {{- $secretName := .Values.existingAirflowSecret }}
-      {{- if .Values.airflow.secretsMapping }}
-        {{- $mapping := .Values.airflow.secretsMapping }}
-      {{- end }}
-    {{- end }}
+    {{- $defaultSecretName := .Release.Name | trunc 63 | trimSuffix "-" }}
+    {{- $secretName := (eq .Values.existingAirflowSecret "") | ternary $defaultSecretName .Values.existingAirflowSecret }}
```

Comment:
Hum is that helm 2.10+ only ? I actually don’t know if we have to maintain backward compatibility. @maver1ck any idea ?

Reply:
Agreed that `ternary` is Helm 2.10+ as per [this commit](https://github.com/helm/helm/commit/aae233e7629e87ed391321df8e7bc0bf66c89371#diff-395f41b2a27b70ce44399c91c82158c2) which bumps the Sprig version. The only reference I can find in the Helm Charts repo to versioning is the short [compatibility blurb](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#compatibility):

> We officially support compatibility with the current and the previous minor version of Kubernetes. Generated resources should use the latest possible API versions compatible with these versions...

But no specific mention of which version of Helm :( We could argue that 2.10 is quite old, though: it's four minor versions behind and was released over [nine months ago](https://github.com/helm/helm/releases/tag/v2.10.0). We could also extend the "current and previous minor version" rule for Kubernetes to apply to Helm as well?

Reply:
Our internal cluster is still under 2.9 sadly that's why I am not very motivated for this change. You however need to update Chart.yaml to describe this requirements (helm>=2.10)